File: grid-pi.py
#!/usr/bin/env python3

acc_in_circle = 0
acc_out_circle = 0

radius = 4000

print('Starting run, radius of %d' % radius)

for x in range (-radius, radius+1):
	for y in range (-radius, radius+1):
		if ((x * x) + (y * y)) > (radius * radius):
			acc_out_circle += 1
		else:
			acc_in_circle += 1

proportion = 4 * (float(acc_in_circle) / (acc_out_circle + acc_in_circle))

print('Run completed')
print('Total points: %d' % (acc_in_circle + acc_out_circle))
print('Inside of circle: %d' % acc_in_circle)
print('Outside of circle: %d' % acc_out_circle)
print('Pi seems to be: %1.10f' % proportion)

File: grid-pi.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $acc_in_circle = 0;
my $acc_out_circle = 0;

my $radius = 4000;

print("Starting run, radius of $radius\n");

for (my $x = ($radius * -1); $x != $radius; $x++) {
	for (my $y = ($radius * -1); $y != $radius; $y++) {
		if ((abs($x) * abs($x)) + (abs($y) * abs($y)) >= ($radius * $radius)) {
			$acc_out_circle++;
		} else {
			$acc_in_circle++;
		}
	}
}

my $proportion = 4.0 * ($acc_in_circle / ($acc_out_circle + $acc_in_circle));

print("Run completed.\n");
print("Total points: " . ($acc_in_circle + $acc_out_circle) . "\n");
print("Inside of circle: $acc_in_circle\n");
print("Outside of circle: $acc_out_circle\n");
print("Pi seems to be... $proportion\n");


